src/Microsoft.AspNetCore.Mvc.Api.Analyzers/AddResponseTypeAttributeCodeFixAction.cs

```diff
@@ -143,20 +157,39 @@ private ICollection<int> CalculateStatusCodesToApply(CodeActionContext context,
                     continue;
                 }
 
-                statusCodes.Add(metadata.IsDefaultResponse ? 200 : metadata.StatusCode);
+                var statusCode = metadata.IsDefaultResponse ? 200 : metadata.StatusCode;
+                statusCodes.Add(statusCode, (statusCode, metadata.ReturnType));
             }
 
-            return statusCodes;
+            return statusCodes.Values;
         }
 
-        private static AttributeSyntax CreateProducesResponseTypeAttribute(CodeActionContext context, int statusCode, out bool addUsingDirective)
+        private static AttributeSyntax CreateProducesResponseTypeAttribute(in CodeActionContext context, int statusCode, out bool addUsingDirective)
         {
+            // [ProducesResponseType(StatusCodes.Status400NotFound)]
             var statusCodeSyntax = CreateStatusCodeSyntax(context, statusCode, out addUsingDirective);
 
             return SyntaxFactory.Attribute(
                 SyntaxFactory.ParseName(ApiSymbolNames.ProducesResponseTypeAttribute)
                     .WithAdditionalAnnotations(Simplifier.Annotation),
                 SyntaxFactory.AttributeArgumentList().AddArguments(
+                    
+                    SyntaxFactory.AttributeArgument(statusCodeSyntax)));
+        }
+
+        private static AttributeSyntax CreateProducesResponseTypeAttribute(in CodeActionContext context, int statusCode, ITypeSymbol typeSymbol, out bool addUsingDirective)
+        {
+            // [ProducesResponseType(typeof(ReturnType), StatusCodes.Status400NotFound)]
+            var statusCodeSyntax = CreateStatusCodeSyntax(context, statusCode, out addUsingDirective);
+            var responseTypeAttribute = SyntaxFactory.TypeOfExpression(
+                SyntaxFactory.ParseTypeName(typeSymbol.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat))
```

Comment:
Consider adding a using static for `SyntaxFactory`. They expect you to use it with using static.

Reply:
I did that at when I started. The problem is that it's difficult to get auto-completion for the methods. I nearly always end up typing `SyntaxFactory.` to get the method list and do the "Simplify name" code fix to fix up the name. Just seems simpler to type out the name.